src/xrpld/app/tx/detail/CreateOffer.cpp

```diff
@@ -950,6 +993,54 @@ CreateOffer::preCompute()
     return Transactor::preCompute();
 }
 
+TER
+CreateOffer::applyHybrid(
+    Sandbox& sb,
+    std::shared_ptr<STLedgerEntry> sleOffer,
+    Keylet const& offerKey,
+    STAmount const& saTakerPays,
+    STAmount const& saTakerGets,
+    std::function<void(SLE::ref, std::optional<uint256>)> const& setDir)
+{
+    if (!sleOffer->isFieldPresent(sfDomainID))
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    // set hybrid flag
+    sleOffer->setFlag(lsfHybrid);
+
+    // if offer is hybrid, need to also place into open offer dir
+    Book const book{saTakerPays.issue(), saTakerGets.issue()};
+
+    auto dir =
+        keylet::quality(keylet::book(book), getRate(saTakerGets, saTakerPays));
+    bool const bookExists = static_cast<bool>(sb.exists(dir));
```

Comment:
static_cast is unnecessary here

Reply:
2c1fe55a887e4199e8b19b050fa6c7c89e95d130